datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaTransportProvider.java

```diff
@@ -80,6 +81,9 @@ public KafkaTransportProvider(DatastreamTask datastreamTask, List<KafkaProducerW
       ErrorLogger.logAndThrowDatastreamRuntimeException(LOG, errorMessage, null);
     }
 
+    _preserveSourceEventTimestamp = Boolean.TRUE.toString().equals(datastreamTask.getDatastreams().get(0).getMetadata()
+            .getOrDefault(DatastreamMetadataConstants.PRESERVE_EVENT_SOURCE_TIMESTAMP, Boolean.FALSE.toString()));
```

Comment:
This can be condensed:

_preserveSourceEventTimestamp = Boolean.parseBoolean(datastreamTask.getDatastreams().get(0).getMetadata().get(DatastreamMetadataConstants. PRESERVE_EVENT_SOURCE_TIMESTAMP));

Reply:
done